lib/spack/spack/ci.py

```diff
@@ -69,48 +72,24 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         return False
 
 
-def get_job_name(spec, osarch, build_group):
-    """Given the necessary parts, format the gitlab job name
+def get_job_name(spec: spack.spec.Spec, format_string: str, build_group: str = ""):
```

Comment:
Can you make sure that this deals with https://docs.gitlab.com/ee/ci/jobs/#job-name-limitations? Truncate on 255 chars, if longer gitlab errors (which we've seen in the past due to long pipeline name :upside_down_face:)